packages/swap/src/routes/v2/quote.ts

```diff
@@ -185,68 +136,56 @@ export const generateQuotes = async ({
 }: {
   dcaQuoteParams?: Awaited<ReturnType<typeof getDcaQuoteParams>>;
   srcAsset: InternalAsset;
-  amount: bigint;
+  depositAmount: bigint;
   destAsset: InternalAsset;
   brokerCommissionBps?: number;
   boostDepositsEnabled: boolean;
   quoter: Quoter;
   autoSlippageEnabled?: boolean;
 }) => {
   const [limitOrders, { estimatedBoostFeeBps, maxBoostFeeBps }, pools] = await Promise.all([
-    quoter.getLimitOrders(srcAsset, destAsset, amount),
+    quoter.getLimitOrders(srcAsset, destAsset, depositAmount),
     env.DISABLE_BOOST_QUOTING || !boostDepositsEnabled
       ? { estimatedBoostFeeBps: undefined, maxBoostFeeBps: undefined }
-      : getBoostFeeBpsForAmount({ amount, asset: srcAsset }),
+      : getBoostFeeBpsForAmount({ amount: depositAmount, asset: srcAsset }),
     getPools(srcAsset, destAsset),
   ]);
 
+  const dcaParams = dcaQuoteParams
+    ? ({
+        numberOfChunks: dcaQuoteParams.numberOfChunks,
+        chunkIntervalBlocks: env.DCA_CHUNK_INTERVAL_BLOCKS,
+      } as DcaParams)
+    : undefined;
+
   const quoteArgs = {
     srcAsset,
     destAsset,
-    depositAmount: amount,
+    depositAmount,
     limitOrders,
     brokerCommissionBps,
     pools,
-    quoteType: 'REGULAR' as const,
-    dcaChunks: 1,
     autoSlippageEnabled,
   };
+  const dcaQuoteArgs = { dcaParams, ...quoteArgs };
 
   const [quoteResult, boostedQuoteResult] = await Promise.allSettled([
     getPoolQuote(quoteArgs),
     estimatedBoostFeeBps && getPoolQuote({ ...quoteArgs, boostFeeBps: estimatedBoostFeeBps }),
   ]);
-  const ingressFee = getFulfilledResult(quoteResult, undefined)?.includedFees.find(
-    (fee) => fee.type === 'INGRESS',
-  );
-
-  // the swap_rate rpc will deduct the full ingress fee before simulating the swap
-  // as we quote a single chunk, we add a surcharge so that the effective deducted amount is 1/numberOfChunks
-  const ingressFeeSurcharge = dcaQuoteParams
-    ? BigInt(
-        new BigNumber(ingressFee?.amount ?? 0)
-          .multipliedBy((dcaQuoteParams.numberOfChunks - 1) / dcaQuoteParams.numberOfChunks)
-          .toFixed(0),
-      )
-    : 0n;
 
   const [dcaQuoteResult, dcaBoostedQuoteResult] = await Promise.allSettled([
     dcaQuoteParams &&
       getPoolQuote({
         ...quoteArgs,
-        depositAmount: dcaQuoteParams.chunkSize + ingressFeeSurcharge,
-        quoteType: 'DCA',
-        dcaChunks: dcaQuoteParams.numberOfChunks,
+        dcaParams,
+      }),
+    dcaQuoteParams &&
+      estimatedBoostFeeBps &&
+      getPoolQuote({
+        ...dcaQuoteArgs,
+        boostFeeBps: estimatedBoostFeeBps,
```

Comment:
should probably check at this point that we have > 1 chunk and not request a quote if we don't

Reply:
could do the check when setting `dcaParams` initially